packages/jest-haste-map/src/index.js

```diff
@@ -437,6 +454,58 @@ class HasteMap {
     }
   }
 
+  _createWatcher(dirPath: string): Promise<Watcher> {
+    const watcher = new Watcher(dirPath, {
+      glob: this._options.extensions.map(extension => `**/*.${extension}`),
+      dot: false,
+    });
+
+    const watcherTimeoutMessage =
+      `Watcher took too long to load (${Watcher.name})` +
+      (
+        Watcher === sane.WatchmanWatcher ?
+        '\nTry running `watchman version` from your terminal' +
+        '\nhttps://facebook.github.io/watchman/docs/troubleshooting.html' :
+        ''
+      );
+
+    return new Promise((resolve, reject) => {
+      const rejectTimeout = setTimeout(
+        () => reject(new Error(watcherTimeoutMessage)),
+        maxWatcherWaitTime,
+      );
+
+      watcher.once('ready', () => {
+        clearTimeout(rejectTimeout);
+        resolve(watcher);
+      });
+    });
+  }
+
+  _startMonitoring(): void {
+    Promise.all(this._options.roots.map(root => this._createWatcher(root)))
+      .then(watchers => {
+        watchers.forEach(watcher => {
+          watcher.on('change', (filepath, root, stat) => {
+            this._rebuild();
+          });
+          watcher.on('add', (filepath, root, stat) => {
+            this._rebuild();
+          });
+          watcher.on('delete', (filepath, root, stat) => {
+            this._rebuild();
+          });
```

Comment:
can we do: 

```
const rebuild = () => this._rebuild();
```

and then reuse it when we attach it to all the events? Seems like a few lines less.